src/spikeinterface/qualitymetrics/misc_metrics.py

```diff
@@ -563,7 +563,129 @@ def compute_synchrony_metrics(waveform_extractor, synchrony_sizes=(2, 4, 8), **k
+            spike_counts, _ = np.histogram(spike_times, bins=edges)
+            firing_rates = spike_counts / bin_size_s
+            firing_rate_histograms[unit_id] = np.concatenate((firing_rate_histograms[unit_id], firing_rates))
+
+    # finally we compute the percentiles
+    firing_ranges = {}
+    for unit_id in unit_ids:
+        firing_ranges[unit_id] = np.percentile(firing_rate_histograms[unit_id], quantiles[1]) - np.percentile(
+            firing_rate_histograms[unit_id], quantiles[0]
+        )
+
+    return firing_ranges
+
+
+_default_params["firing_range"] = dict(bin_size_s=5, quantiles=(0.05, 0.95))
+
+
+def compute_amplitude_spreads(
+    waveform_extractor, num_spikes_per_bin=100, amplitude_extension="spike_amplitudes", unit_ids=None
+):
+    """Calculate spread of spike amplitudes within defined bins of spike events.
+    The spread is the median relative variance (variance divided by the overall amplitude mean)
+    computed over bins of `num_spikes_per_bin` spikes.
+
+    Parameters
+    ----------
+    waveform_extractor : WaveformExtractor
+        The waveform extractor object.
+    num_spikes_per_bin : int, default: 50
+        The number of spikes per bin.
+    amplitude_extension : str, default: 'spike_amplitudes'
+        The name of the extension to load the amplitudes from. 'spike_amplitudes' or 'amplitude_scalings'.
+    unit_ids : list or None
+        List of unit ids to compute the amplitude spread. If None, all units are used.
+
+    Returns
+    -------
+    amplitude_spreads : dict
+        The amplitude spread for each unit.
+
+    Notes
+    -----
+    Designed by Simon Musall and ported to SpikeInterface by Alessio Buccino.
+    """
+    assert amplitude_extension in (
+        "spike_amplitudes",
+        "amplitude_scalings",
+    ), "Invalid amplitude_extension. It can be either 'spike_amplitudes' or 'amplitude_scalings'"
+    sorting = waveform_extractor.sorting
+    spikes = sorting.to_spike_vector()
+    num_spikes = sorting.count_num_spikes_per_unit()
+    if unit_ids is None:
+        unit_ids = sorting.unit_ids
+
+    if waveform_extractor.is_extension(amplitude_extension):
+        sac = waveform_extractor.load_extension(amplitude_extension)
+        amps = sac.get_data(outputs="concatenated")
+        if amplitude_extension == "spike_amplitudes":
+            amps = np.concatenate(amps)
+    else:
+        warnings.warn("")
+        empty_dict = {unit_id: np.nan for unit_id in unit_ids}
+        return empty_dict
+
+    all_unit_ids = list(sorting.unit_ids)
+    amplitude_spreads = {}
+    for unit_id in unit_ids:
+        amps_unit = amps[spikes["unit_index"] == all_unit_ids.index(unit_id)]
+        amp_mean = np.abs(np.mean(amps_unit))
+        if num_spikes[unit_id] < num_spikes_per_bin:
+            amp_spread = np.std(amps_unit) / amp_mean
+        else:
+            amp_spreads = []
+            for i in range(0, num_spikes[unit_id], num_spikes_per_bin):
```

Comment:
I am not sure this is a good idea to have bins that are number of spikes.
It would be more intuituive to have real temporal bins.
The bin will be really diffrents depending across units the firing rate.

Reply:
The idea for using an equal number of spikes was to make sure that the variance in the amplitudes is not driven by variability in firing rate. There are sometimes bursts of high activity and then episodes in the recording with lower activity. When we tried a time-based binning approach this variability in the number of spikes per bin was driving the result and we also had to exclude bins with no spikes.

Reply:
@musall actually even using a fixed number of spikes would suffer from variability in firing range. Potentially even more tha using temporal bins. 

Let's take an edge case: let's assume that `num_spikes_per_bin=50` and a neuron fires a 50-spikes burst in 100ms. In this case, the amplitude variance over 50 spikes will be **strongly** impacted by the firing rate. Instead, if we used constant temporal bins (say 10s) and the neuron went back to normal firing regime, the variance would be better estimated. What do you think?

Reply:
Maybe we could auto-estimate a correct bin size, based on the average firing rate? E.g. if firing rate is 1Hz and the average number of spikes in bins is 50, then the temporal bin size should be 50 seconds.

Reply:
That is a good idea. If we can infer temporal bins that are appropriate for the firing rate of a given neuron that should be more stable. We can also test different implementations against human labels to see how well they predict different clusters. We just used an AUC for that and could test that easily.
Would you just estimate the temporal bin size to contain a specific number of spikes on average?